### DISCUSSION TITLE
I can't pass parameters to the component.

### DISCUSSION THREAD

**Discussion by chrisklein00** [wire-elements/modal#204] [Q&A] [answered]
I can't pass parameters to the component.

I have 3 files:

Inside Livewire/test.blade.php
`onclick='Livewire.emit("openModal", "test-modal", {{ json_encode(["id_code" => $client->id_code]) }})'>`

Inside /Http/Livewire/TestModal.php
--------------

```
namespace App\Http\Livewire;

use LivewireUI\Modal\ModalComponent;
use App\Models\Client;


class TestModal extends ModalComponent
{
    public $id_code;

    public function render($id_code)
    {
        dd($id_code);
        return view('livewire.test-modal');
    }
}
```

And livewire.test-modal which displays the content of the modal window.

But I can't get the id_code.

Let's see if someone can help me with this. Thanks.

**Comment by lukedeboer** [comment, marked as answer]
Aren't you supposed to be using the mount method to accept them?

**Reply by PhiloNL** [reply]
This is correct, the mount method should be used.
